src/components/molecules/UiLoader/UiLoader.vue

```diff
@@ -89,56 +90,108 @@ const props = defineProps({
     default: 'div',
   },
   /**
-   * Use this props to pas transition name
+   * Use this props to set transition name
    */
-  transition: {
-    type: [String, Boolean] as PropType<string | boolean>,
-    default: false,
+  name: {
+    type: String,
+    default: 'v',
   },
   /**
-   * Use this props to start rendering final component before loading stops, so it's completely rendered
+   * Use this props to apply a transition on the initial render
    */
-  eagerLoadComponent: {
+  appear: {
     type: Boolean,
-    default: false,
+    default: true,
   },
 });
-const component = computed<LoaderComponent>(() => {
+const loaderEl = ref<HTMLElement | null>(null);
+const contentEl = ref<HTMLElement | null>(null);
+const loaderComponent = computed<LoaderComponent>(() => {
   const components: Record<LoaderType, LoaderComponent> = {
     spinner: UiLoaderSpinner,
     ellipsis: UiLoaderEllipsis,
     skeleton: UiLoaderSkeleton,
   };
   return components[props.type];
 });
-const outerComponent = computed<VNode>(() => (
-  props.transition
-    ? h(TransitionGroup, { name: props.transition as string })
-    : h((_, { slots }) => (slots.default as Slot)())
-));
+const animateEl = ref<'loader' | 'content'>(props.isLoading ? 'loader' : 'content');
+const isLoaderAnimate = computed(() => animateEl.value === 'loader');
+const isContentAnimate = computed(() => animateEl.value === 'content');
+const transitionState = reactive({
+  from: false,
+  to: false,
+});
+const loaderClass = computed(() => {
+  const { from, to } = transitionState;
+  const move = props.isLoading ? 'enter' : 'leave';
+  return {
+    [`${props.name}-${move}-from`]: from && isLoaderAnimate.value,
+    [`${props.name}-${move}-active`]: to && isLoaderAnimate.value,
+    [`${props.name}-${move}-to`]: to && isLoaderAnimate.value,
+  };
+});
+const contentClass = computed(() => {
+  const { from, to } = transitionState;
+  const move = !props.isLoading ? 'enter' : 'leave';
+  return {
+    [`${props.name}-${move}-from`]: from && isContentAnimate.value,
+    [`${props.name}-${move}-active`]: to && isContentAnimate.value,
+    [`${props.name}-${move}-to`]: to && isContentAnimate.value,
+  };
+});
+const loaderStyle = computed<CSSProperties>(() => ({
+  visibility: isLoaderAnimate.value ? undefined : 'hidden',
+  position: isLoaderAnimate.value && props.type !== 'ellipsis' ? undefined : 'absolute',
+}));
+const contentStyle = computed<CSSProperties>(() => ({
+  visibility: isContentAnimate.value ? undefined : 'hidden',
+  position: isContentAnimate.value || props.type === 'ellipsis' ? undefined : 'absolute',
+}));
```

Comment:
I will think how to optimize your solution on Tuesday. We do the same things for loader and content. Maybe is some way to move it to one hmm "object" and only switch between loader/content element.

Reply:
So, yep, there are two almost identical blocks of code I can distinguish; these are:

```js
const isLoaderAnimate = computed(() => animateEl.value === 'loader');
const loaderClass = computed(() => {
  const { from, to } = transitionState;
  const move = props.isLoading ? 'enter' : 'leave';
  return {
    [`${props.name}-${move}-from`]: from && isLoaderAnimate.value,
    [`${props.name}-${move}-active`]: to && isLoaderAnimate.value,
    [`${props.name}-${move}-to`]: to && isLoaderAnimate.value,
  };a
});
const loaderStyle = computed<CSSProperties>(() => ({
  visibility: isLoaderAnimate.value ? undefined : 'hidden',
  position: isLoaderAnimate.value && props.type !== 'ellipsis' ? undefined : 'absolute',
}));
```

and

```ts
const isContentAnimate = computed(() => animateEl.value === 'content');
const contentClass = computed(() => {
  const { from, to } = transitionState;
  const move = !props.isLoading ? 'enter' : 'leave';
  return {
    [`${props.name}-${move}-from`]: from && isContentAnimate.value,
    [`${props.name}-${move}-active`]: to && isContentAnimate.value,
    [`${props.name}-${move}-to`]: to && isContentAnimate.value,
  };
});
const contentStyle = computed<CSSProperties>(() => ({
  visibility: isContentAnimate.value ? undefined : 'hidden',
  position: isContentAnimate.value || props.type === 'ellipsis' ? undefined : 'absolute',
}));
```
So I thought, why not create some function to re-use this functionality, maybe something like this:

```ts
const useAnimate = (isAnimate, move, pAbsoluteCondition) => {
  const elementClass = computed(() => {
    const { from, to } = transitionState;
    return {
      [`${props.name}-${move}-from`]: from && isAnimate.value,
      [`${props.name}-${move}-active`]: to && isAnimate.value,
      [`${props.name}-${move}-to`]: to && isAnimate.value,
    };
  });
  const elementStyle = computed<CSSProperties>(() => ({
    visibility: isAnimate.value ? undefined : 'hidden',
    position: pAbsoluteCondition ? undefined : 'absolute',
  }));

  return { elementClass, elementStyle };
}
```

What do you think?